packages/svelte/src/internal/client/runtime.js

```diff
@@ -239,6 +242,74 @@ export function check_dirtiness(reaction) {
 	return is_dirty;
 }
 
+/**
+ * @param {Error} error
+ * @param {import("#client").Effect} effect
+ * @param {import("#client").ComponentContext | null} component_context
+ */
+function handle_error(error, effect, component_context) {
+	// Given we don't yet have error boundaries, we will just always throw.
+	if (!DEV || handled_errors.has(error) || component_context === null) {
+		throw error;
+	}
+
+	const component_stack = [];
+
+	/** @type {import("#client").ComponentContext | null} */
+	let current_context = component_context;
+
+	while (current_context !== null) {
+		var filename = current_context.function?.filename;
+
+		if (filename) {
+			const file = filename.split('/').at(-1);
+			component_stack.push(file);
+		}
+
+		current_context = current_context.p;
+	}
+	const effect_name = effect.fn.name;
+
+	if (effect_name) {
+		// If we have an effect name, it means it happened in the template, $effect or $effect.pre.
+		Object.defineProperty(error, '__skip_hydration_retry', {
+			value: true,
+			enumerable: false
+		});
+	}
```

Comment:
this heuristic feels brittle. if an error occurs during component initialisation, `effect_name` is empty, and `skip_retry` will be false. AFAICT there's only one way a hydration mismatch error will be thrown, and that's `e.hydration_missing_marker_close()` in `hydration.js`. What if we just check whether the caught error had that code, rather than this `__skip_hydration_retry` stuff?

Reply:
Agreed

Reply:
Are you sure this is the only way this can happen? What if the DOM was modified by some extension and a bogus element was inserted, so the walking mechanism fails and a delete everything + remount would fix it?

Reply:
> delete everything + remount would fix it?

That's not really a fix though, remounting everything is a last resort and they should really fix their problem, no?

Reply:
What would they be able to fix if a browser extension messes with the dom?

Reply:
Browser extensions break every framework, regardless of us trying to fix it during hydration. This has plagued every library/framework since the dawn of time too. I remember having issues with this and jQuery + ad blocker back in the day!

This was particularly complex with Lexical because of contentEditable. The only thing you can do is let people complain and then point them to turning off their extension. You can go further and add a MutationObserved in DEV that warns on mutations outside of Svelte, but that seems like a step too far for us to go.

Reply:
this seems like overkill to me (and what if the element was wrong but had the same name? we'd still be screwed)